### PR TITLE
Create audit readme

### PR DESCRIPTION
**Pull request by alcueca**
Please check if you can add anything.

Color on how things work the way they do is useful for auditors to get a picture of the contract. If I get time we can put the formula for the Dutch auction, or some diagram of token and control flows.

### CODE REVIEW

File: packages/foundry/contracts/README_WITCH.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
+# Auditing the Witch v2
+This contest is for the first major refactor on liquidations engine for the Yield Protocol. The existing engine has been heavily used since its launch in October 2021 and in that time we have found that liquidations have not been as fair as we would have liked. More often than not, liquidated users have lost all their collateral as we have failed to make liquidations competitive.
+
+This second version of the Witch aims to:
+1. Give us more flexibility in exploring different liquidation models, including moving away from Dutch auctions to fixed discount if there is not enough competition from liquidators.
```
Comment (bbonanno) [inline packages/foundry/contracts/README_WITCH.md:5]: I'd rephrase to "including starting auctions at a fixed discount so most of the times vaults can be liquidated in a single transaction" or something like that.
My point is, we're not really moving away of the auctions.

File: packages/foundry/contracts/README_WITCH.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+There are exceptions to this rule.
+1. It is possible to set any given collateral/underlying pair as out of scope for the Witch. Some pairs should not be liquidated (such as DAI/fyDAI).
+2. It is possible to set any owner to be protected from liquidations. The goal for this is to be able to run parallel liquidation engines.
+
+Governance can choose whether liquidations for any pair should be of the whole vault, or of only a part of the vault.
+
+Auctions are by default a reverse Dutch auction, in which liquidators offer to pay part or all of the debt in the vault, and in exchange they get an amount of the vault collateral that increases with time. At the end of an auction a liquidator gets all the collateral under auction in exchange for paying all the debt under auction. It is possible for governance to set the amount of collateral that would be given for paying all the debt at the start of the auction, and the length of the auction. By modifying these two parameters governance can choose any linera progression for the auction.
```
Comment (bbonanno) [inline packages/foundry/contracts/README_WITCH.md:23]: *linear

File: packages/foundry/contracts/README_WITCH.md (hunk 1 above)
Comment (bbonanno) [reply packages/foundry/contracts/README_WITCH.md]: Yes, but that's orthogonal to this, maybe tomorrow we can have a call ?

Comment (alcueca) [reply packages/foundry/contracts/README_WITCH.md]: It's not orthogonal, part of the purpose of this refactor was to enable us to experiment with Compound-style liquidations. [Check this PR](https://github.com/yieldprotocol/vault-v2/pull/466).

Comment (bbonanno) [reply packages/foundry/contracts/README_WITCH.md]: It is, maybe we're talking about the same thing (or not), that's why I think it's easier to discuss it on zoom :)

File: packages/foundry/contracts/README_WITCH.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+Governance can choose whether liquidations for any pair should be of the whole vault, or of only a part of the vault.
+
+Auctions are by default a reverse Dutch auction, in which liquidators offer to pay part or all of the debt in the vault, and in exchange they get an amount of the vault collateral that increases with time. At the end of an auction a liquidator gets all the collateral under auction in exchange for paying all the debt under auction. It is possible for governance to set the amount of collateral that would be given for paying all the debt at the start of the auction, and the length of the auction. By modifying these two parameters governance can choose any linear progression for the auction.
+
+The Witch can be configured to sell collateral under liquidation at a fixed discount. For that we would set the duration to 2^32 - 1, interpreted as infinite. When configured that way, the auction price doesn't change with time and is only determined by the configured `initialOffer` parameter. This is a similar behaviour to Compound liquidations.
```
Comment (bbonanno) [inline packages/foundry/contracts/README_WITCH.md:28]: I strongly disagree on this one

File: packages/foundry/contracts/README_WITCH.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+Governance can choose whether liquidations for any pair should be of the whole vault, or of only a part of the vault.
+
+Auctions are by default a reverse Dutch auction, in which liquidators offer to pay part or all of the debt in the vault, and in exchange they get an amount of the vault collateral that increases with time. At the end of an auction a liquidator gets all the collateral under auction in exchange for paying all the debt under auction. It is possible for governance to set the amount of collateral that would be given for paying all the debt at the start of the auction, and the length of the auction. By modifying these two parameters governance can choose any linera progression for the auction.
+
+A special auction setting would be to set the duration to zero seconds. When configured that way, there is no auction at all and all the collateral under auction is immediately offered for all the debt under auction. In combination with auctioning only part of a vault this has a similar behaviour to Compound liquidations without a penalty fee.
```
[the remark this answers is not in the document]
Comment (alcueca) [reply packages/foundry/contracts/README_WITCH.md]: Removed

File: packages/foundry/contracts/README_WITCH.md (hunk 3 above)
Comment (alcueca) [reply packages/foundry/contracts/README_WITCH.md]: Modified, as discussed.

**Comment by alcueca** [comment]
Moved into docs-v2
